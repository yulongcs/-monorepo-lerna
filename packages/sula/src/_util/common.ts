import assignWith from 'lodash/assignWith';
import isUndefined from 'lodash/isUndefined';
import isArray from 'lodash/isArray';

function withDefined(dest: any, src: any) {
  return isUndefined(src) ? dest : src;
}

export function assignWithDefined(...args: Record<string, any>[]) {
  return assignWith(...args, withDefined);
}

export function toArray<T>(value?: T | T[] | null): T[] {
  if (value === undefined || value === null) {
    return [];
  }

  // @ts-ignore
  return isArray(value) ? value : [value];
}


