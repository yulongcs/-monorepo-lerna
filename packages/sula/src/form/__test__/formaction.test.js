import React from 'react';
import { mount } from 'enzyme';
import { Form, FormAction } from '..';
import '../../__tests__/common';

describe('form action', () => {
  it('formaction in form', () => {
    const fn = jest.fn();
    const wrapper = mount(
      <Form
        fields={[
          {
            name: 'group',
            actionsRender: [
              {
                type: 'button',
                props: {
                  className: 'action-test-btn',
                  children: 'btn',
                },
                action: fn,
              },
            ],
            actionsPosition: 'center',
            fields: [
              {
                name: 'input',
                label: 'input',
                field: 'input',
              },
            ],
          },
        ]}
      />,
    );

    wrapper.find('.action-test-btn').last().simulate('click');
    expect(fn.mock.calls[0][0].form).not.toBeNull();
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('formaction component', () => {
    const fn = jest.fn();
    const wrapper = mount(
      <Form layout="vertical">
        <FormAction
          actionsRender={[
            {
              type: 'button',
              props: {
                className: 'action-test-btn',
                children: 'btn',
              },
              action: fn,
            },
          ]}
        />
      </Form>,
    );
    wrapper.find('.action-test-btn').last().simulate('click');
    expect(fn.mock.calls[0][0].form).not.toBeNull();
    expect(wrapper.render()).toMatchSnapshot();
  });
});
