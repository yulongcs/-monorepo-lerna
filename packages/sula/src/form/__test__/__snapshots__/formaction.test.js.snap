// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form action formaction component 1`] = `
<form
  class="ant-form ant-form-vertical"
>
  <div
    class="ant-row ant-form-item"
    style="margin-bottom: 0px;"
  >
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <div
            class="sula-form-action-default"
          >
            <button
              class="ant-btn action-test-btn"
              type="button"
            >
              <span>
                btn
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`form action formaction in form 1`] = `
<form
  class="ant-form ant-form-horizontal"
>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-col ant-col-8 ant-form-item-label"
    >
      <label
        class=""
        for="input"
        title="input"
      >
        input
      </label>
    </div>
    <div
      class="ant-col ant-col-12 ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <input
            class="ant-input"
            id="input"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="sula-form-action-center"
  >
    <button
      class="ant-btn action-test-btn"
      type="button"
    >
      <span>
        btn
      </span>
    </button>
  </div>
</form>
`;
